frontend/src/components/ui/data-grid/data-grid-row.ts

```diff
@@ -0,0 +1,383 @@
+    console.debug("form associated");
+  }
+
+  public formResetCallback() {
+    this.setValue(this.item || {});
+    this.commitValue();
+  }
+
+  public formDisabledCallback(disabled: boolean) {
+    console.debug("form disabled:", disabled);
+  }
+
+  public formStateRestoreCallback(state: string | FormData, reason: string) {
+    console.debug("formStateRestoreCallback:", state, reason);
+  }
+
+  public checkValidity(): boolean {
+    return this.#internals.checkValidity();
+  }
+
+  public reportValidity(): void {
+    this.#internals.reportValidity();
+  }
+
+  public get validity(): ValidityState {
+    return this.#internals.validity;
+  }
+
+  public get validationMessage(): string {
+    return this.#internals.validationMessage;
+  }
+
+  constructor() {
+    super();
+    this.#internals = this.attachInternals();
+  }
+
+  protected createRenderRoot() {
+    const root = super.createRenderRoot();
+
+    // Attach to render root so that `e.target` is table cell
+    root.addEventListener(
+      "btrix-input",
+      (e) => void this.onCellInput(e as CustomEvent<CellEditEventDetail>),
+    );
+    root.addEventListener(
+      "btrix-change",
+      (e) => void this.onCellChange(e as CustomEvent<CellEditEventDetail>),
+    );
+
+    return root;
+  }
+
+  protected willUpdate(changedProperties: PropertyValues): void {
+    if (
+      (changedProperties.has("item") || changedProperties.has("editCells")) &&
+      this.item &&
+      this.editCells
+    ) {
+      this.setValue(this.item);
+      this.commitValue();
+    }
+  }
+
+  @queryAll("btrix-data-grid-cell")
+  private readonly gridCells?: NodeListOf<DataGridCell>;
+
+  private setValue(cellValues: Partial<GridItem>) {
+    Object.keys(cellValues).forEach((field) => {
+      this.cellValues[field] = cellValues[field];
+    });
+
+    this.#internals.setFormValue(JSON.stringify(this.cellValues));
+  }
+
+  private commitValue() {
+    this.cellValues = {
+      ...this.cellValues,
+    };
+  }
+
+  render() {
+    if (!this.columns?.length) return html``;
+
+    let removeCell = html``;
+
+    if (this.removable) {
+      removeCell = html`
+        <btrix-data-grid-cell
+          class=${clsx(tw`border-l p-0`, cellStyle)}
+          @keydown=${this.onKeydown}
+        >
+          <sl-tooltip content=${msg("Remove")}>
+            <sl-icon-button
+              class="p-1 text-base hover:text-danger"
+              name="trash3"
+              @click=${() =>
+                this.dispatchEvent(
+                  new CustomEvent<RowRemoveEventDetail>("btrix-remove", {
+                    detail: {
+                      key: this.key,
+                    },
+                    bubbles: true,
+                    composed: true,
+                  }),
+                )}
+            ></sl-icon-button>
+          </sl-tooltip>
+        </btrix-data-grid-cell>
+      `;
+    }
+
+    return html` ${this.columns.map(this.renderCell)} ${removeCell} `;
```

Comment:
_Extremely_ minor nit, but I don't think the extra whitespaces are necessary
```suggestion
    return html`${this.columns.map(this.renderCell)}${removeCell}`;
```

Reply:
Ah yeah I think this was from auto-formatting